shell/common/shell_unittests.cc

```diff
@@ -3339,12 +3339,7 @@ TEST_F(ShellTest, ImageGeneratorRegistryNotNullAfterParentShellDestroyed) {
 TEST_F(ShellTest, UpdateAssetResolverByTypeReplaces) {
   ASSERT_FALSE(DartVMRef::IsInstanceRunning());
   Settings settings = CreateSettingsForFixture();
-  ThreadHost thread_host("io.flutter.test." + GetCurrentTestName() + ".",
-                         ThreadHost::Type::Platform);
-  auto task_runner = thread_host.platform_thread->GetTaskRunner();
-  TaskRunners task_runners("test", task_runner, task_runner, task_runner,
-                           task_runner);
-  auto shell = CreateShell(std::move(settings), task_runners);
+  auto shell = CreateShell(std::move(settings));
```

Comment:
I think we still want this test to pass as is. Instead of modifying the threading configuration here, can you add a new test?

If you use `RunNowOrPostTask` this test should be fine (although it may need to update some expectations below).

Reply:
I have modified based on your comments, but I still have a question for this test case,  since it is not real that the four flutter threads(UI/raster/IO/Platform) run in just one thread, why not create four threads for each?

Reply:
An embedder can choose to run single threaded, and even in "normal" Flutter applications threads might get merged (although not normally the UI thread).